AddCommand with invalid date of birth does not show error on UI

When adding invalid date of birth (e.g. 18 Jun 2432) using `add` command, the UI neither shows error nor does it add the person.

Command which was run: `add n/John Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 dob/18 Jun 2432 pri/HIGH r/Promises to pay back next week. t/friends t/owesMoney`.

On the terminal, an exception showed up: `Caused by: java.lang.IllegalArgumentException: Date of birth must not be a future date.`.

--- a/build.gradle
+++ b/build.gradle
@@ -20,6 +20,10 @@ checkstyle {
     toolVersion = '10.2'
 }
 
+run {
+    enableAssertions = true
+}
+
 test {
     useJUnitPlatform()
     finalizedBy jacocoTestReport

--- a/src/main/java/seedu/address/logic/parser/ParserUtil.java
+++ b/src/main/java/seedu/address/logic/parser/ParserUtil.java
@@ -112,7 +112,7 @@ public class ParserUtil {
     public static DateOfBirth parseDateOfBirth(String dateOfBirth) throws ParseException {
         requireNonNull(dateOfBirth);
         String trimmedDate = dateOfBirth.trim();
-        if (!Date.isValidDate(trimmedDate)) {
+        if (!DateOfBirth.isValidDate(trimmedDate)) {
             throw new ParseException(Date.MESSAGE_CONSTRAINTS);
         }
         return new DateOfBirth(trimmedDate);

--- a/src/main/java/seedu/address/logic/parser/ParserUtil.java
+++ b/src/main/java/seedu/address/logic/parser/ParserUtil.java
@@ -13,7 +13,6 @@ import seedu.address.commons.core.index.Index;
 import seedu.address.commons.util.StringUtil;
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.person.Address;
-import seedu.address.model.person.Date;
 import seedu.address.model.person.DateOfBirth;
 import seedu.address.model.person.Email;
 import seedu.address.model.person.Income;
@@ -112,8 +111,9 @@ public class ParserUtil {
     public static DateOfBirth parseDateOfBirth(String dateOfBirth) throws ParseException {
         requireNonNull(dateOfBirth);
         String trimmedDate = dateOfBirth.trim();
+
         if (!DateOfBirth.isValidDate(trimmedDate)) {
-            throw new ParseException(Date.MESSAGE_CONSTRAINTS);
+            throw new ParseException(DateOfBirth.MESSAGE_CONSTRAINTS);
         }
         return new DateOfBirth(trimmedDate);
     }

--- a/src/main/java/seedu/address/model/person/DateOfBirth.java
+++ b/src/main/java/seedu/address/model/person/DateOfBirth.java
@@ -4,7 +4,6 @@ import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
 import java.time.LocalDate;
-import java.time.format.DateTimeParseException;
 
 /**
  * A class to represent a person's date of birth in the address book.
@@ -12,7 +11,8 @@ import java.time.format.DateTimeParseException;
  */
 public class DateOfBirth extends Date {
 
-    public static final String MESSAGE_CONSTRAINTS = "Date of birth must not be a future date.";
+    public static final String MESSAGE_CONSTRAINTS =
+            String.format("%s. Date of birth must not be a future date.", Date.MESSAGE_CONSTRAINTS);
 
     /**
      * Constructs a {@code DateOfBirth}.
@@ -33,12 +33,12 @@ public class DateOfBirth extends Date {
     public static boolean isValidDate(String test) {
         requireNonNull(test);
 
-        try {
-            LocalDate localDate = Date.parseLocalDate(test);
-            LocalDate now = LocalDate.now();
-            return localDate.isBefore(now) || localDate.isEqual(now);
-        } catch (DateTimeParseException e) {
+        if (!Date.isValidDate(test)) {
             return false;
         }
+
+        LocalDate localDate = Date.parseLocalDate(test);
+        LocalDate now = LocalDate.now();
+        return localDate.isBefore(now) || localDate.isEqual(now);
     }
 }

--- a/src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
+++ b/src/test/java/seedu/address/logic/parser/AddCommandParserTest.java
@@ -50,7 +50,7 @@ import org.junit.jupiter.api.Test;
 import seedu.address.logic.Messages;
 import seedu.address.logic.commands.AddCommand;
 import seedu.address.model.person.Address;
-import seedu.address.model.person.Date;
+import seedu.address.model.person.DateOfBirth;
 import seedu.address.model.person.Email;
 import seedu.address.model.person.Name;
 import seedu.address.model.person.Person;
@@ -223,7 +223,7 @@ public class AddCommandParserTest {
         // invalid date of birth
         assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
                 + INVALID_DATE_OF_BIRTH_DESC + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
-                Date.MESSAGE_CONSTRAINTS);
+                DateOfBirth.MESSAGE_CONSTRAINTS);
 
         // invalid tag
         assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB

--- a/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
+++ b/src/test/java/seedu/address/logic/parser/ParserUtilTest.java
@@ -37,6 +37,7 @@ public class ParserUtilTest {
     private static final String INVALID_TAG = "#friend";
     private static final String INVALID_DATE_1 = "129 March 00021";
     private static final String INVALID_DATE_2 = "zxcvbn";
+    private static final String INVALID_DATE_FUTURE = "19 Mar 2900";
     private static final String INVALID_INCOME = "-21092";
 
     private static final String VALID_NAME = "Rachel Walker";
@@ -309,6 +310,11 @@ public class ParserUtilTest {
     public void parseDateOfBirth_invalidValue_throwsParseException() {
         assertThrows(ParseException.class, () -> ParserUtil.parseDateOfBirth(INVALID_DATE_1));
         assertThrows(ParseException.class, () -> ParserUtil.parseDateOfBirth(INVALID_DATE_2));
+    }
+
+    @Test
+    public void parseDateOfBirth_futureDate_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseDateOfBirth(INVALID_DATE_FUTURE));
     }
 
     @Test

